### PR TITLE
Adding jove

### LINKED ISSUE #10904: package jove
As described nicely in the following article, jove (Jonathan's own version of Emacs) is a lightweight alternative to emacs:

https://opensource.com/article/17/1/jove-lightweight-alternative-vim

Debian and Fedora packages:

https://packages.debian.org/search?keywords=jove
https://apps.fedoraproject.org/packages/jove

Upstream:

ftp://ftp.cs.toronto.edu/pub/hugh/jove-dev/

Current latest release is Jove v4.16.0.73 from July 2010,

ftp://ftp.cs.toronto.edu/pub/hugh/jove-dev/jove4.16.0.73.tgz

There is an experimental release v4.17.0.74 from October 2015,

ftp://ftp.cs.toronto.edu/pub/hugh/jove-dev/experimental/jove4.16.0.74.tgz

### PR DESCRIPTION
**Pull request by peterjc**
Aims to close #10904 

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by peterjc** [comment]
The macOS failure is consistent with my local attempt:

```
x86_64-apple-darwin13.4.0-clang -O  -DLinux -D_FORTIFY_SOURCE=2 -isystem $PREFIX/include -mmacosx-version-min=10.9  -c -o jove.o jove.c
jove.c:1829:23: error: use of undeclared identifier 'SIGWINCH'
        (void) setsighandler(SIGWINCH, win_reshape);
                             ^
1 error generated.
```

Might have to remove that ``-DLinux`` bit...

**Comment by peterjc** [comment]
Compiles now on macOS (but not Linux or Windows), looks like $PREFIX isn't picked up automatically. Might patch ``JOVEHOME = $PREFIX`` to solve that?

**Comment by peterjc** [comment]
Linux failing to link, looks like I have not got the ncurses dependency right:

```
...
$BUILD_PREFIX/bin/x86_64-conda-linux-gnu-cc  -O  -o jjove commands.o keys.o argcount.o ask.o buf.o jctype.o delete.o disp.o insert.o io.o jove.o marks.o misc.o re.o screen.o termcap.o unix.o util.o vars.o list.o keymaps.o mouse.o jtc.o abbrev.o rec.o para.o fmt.o c.o wind.o fp.o move.o extend.o macros.o iproc.o reapp.o proc.o scandir.o term.o case.o -lncurses 
/home/conda/staged-recipes/build_artifacts/jove_1611244130497/_build_env/bin/../lib/gcc/x86_64-conda-linux-gnu/9.3.0/../../../../x86_64-conda-linux-gnu/bin/ld: cannot find -lncurses
collect2: error: ld returned 1 exit status
```

**Comment by peterjc** [comment]
Mac OS X aka macOS - passed.

Linux - failing to link to ncurses.

Windows - failing to find compilers?

**Comment by peterjc** [comment]
Linux still failing with ``ld: cannot find -lncurses`` despite adding ``TERMCAPLIB=-lncurses`` to the make call:

```
...
$BUILD_PREFIX/bin/x86_64-conda-linux-gnu-cc  -O  -o jjove commands.o keys.o argcount.o ask.o buf.o jctype.o delete.o disp.o insert.o io.o jove.o marks.o misc.o re.o screen.o termcap.o unix.o util.o vars.o list.o keymaps.o mouse.o jtc.o abbrev.o rec.o para.o fmt.o c.o wind.o fp.o move.o extend.o macros.o iproc.o reapp.o proc.o scandir.o term.o case.o -lncurses 
/home/conda/staged-recipes/build_artifacts/jove_1611308863320/_build_env/bin/../lib/gcc/x86_64-conda-linux-gnu/9.3.0/../../../../x86_64-conda-linux-gnu/bin/ld: cannot find -lncurses
collect2: error: ld returned 1 exit status
make: *** [jjove] Error 1
```

**Comment by peterjc** [comment]
I may have to disable building for 64 bit Windows pending https://github.com/jonmacs/jove/issues/4

**Comment by peterjc** [comment]
Hmm, adding ``-ltinfo`` for Linux just changed the error to:

```
...
$BUILD_PREFIX/bin/x86_64-conda-linux-gnu-cc  -O  -o jjove commands.o keys.o argcount.o ask.o buf.o jctype.o delete.o disp.o insert.o io.o jove.o marks.o misc.o re.o screen.o termcap.o unix.o util.o vars.o list.o keymaps.o mouse.o jtc.o abbrev.o rec.o para.o fmt.o c.o wind.o fp.o move.o extend.o macros.o iproc.o reapp.o proc.o scandir.o term.o case.o -lncurses -ltinfo 
/home/conda/staged-recipes/build_artifacts/jove_1611319452452/_build_env/bin/../lib/gcc/x86_64-conda-linux-gnu/9.3.0/../../../../x86_64-conda-linux-gnu/bin/ld: cannot find -lncurses
/home/conda/staged-recipes/build_artifacts/jove_1611319452452/_build_env/bin/../lib/gcc/x86_64-conda-linux-gnu/9.3.0/../../../../x86_64-conda-linux-gnu/bin/ld: cannot find -ltinfo
collect2: error: ld returned 1 exit status
make: *** [jjove] Error 1
```

Guess the problem is LDFLAGS instead.

**Comment by peterjc** [comment]
Linux: OK

MacOS: OK

Windows is failing somewhere in https://github.com/conda-forge/vc-feedstock/blob/master/recipe/activate.bat from the error messages?

**Comment by peterjc** [comment]
Windows does not seem to get as far as my build command with make:

```
...
Windows SDK version found as: "10.0.17763.0"

%SRC_DIR%>IF "win-64" == "win-64" (
set "CMAKE_GEN=Visual Studio 15 2017 Win64"  
 set "BITS=64" 
)  else (
set "CMAKE_GEN=Visual Studio 15 2017"  
 set "BITS=32" 
) 

%SRC_DIR%>pushd C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\ 

C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise>CALL "VC\Auxiliary\Build\vcvars64.bat" -vcvars_ver=14.16 10.0.17763.0 
**********************************************************************
** Visual Studio 2017 Developer Command Prompt v15.9.30
** Copyright (c) 2017 Microsoft Corporation
**********************************************************************
[vcvarsall.bat] Environment initialized for: 'x64'
  File ".ci_support\build_all.py", line 157, in <module>
    build_all(os.path.join(root_dir, "recipes"), args.arch)
  File ".ci_support\build_all.py", line 81, in build_all
    build_folders(recipes_dir, new_comp_folders, arch, channel_urls)
  File ".ci_support\build_all.py", line 138, in build_folders
    conda_build.api.build([recipe], config=get_config(arch, channel_urls))
  File "C:\Miniconda\lib\site-packages\conda_build\api.py", line 195, in build
    variants=variants
  File "C:\Miniconda\lib\site-packages\conda_build\build.py", line 3073, in build_tree
    notest=notest,
  File "C:\Miniconda\lib\site-packages\conda_build\build.py", line 2159, in build
    windows.build(m, build_file, stats=build_stats, provision_only=provision_only)
  File "C:\Miniconda\lib\site-packages\conda_build\windows.py", line 299, in build
    check_call_env(cmd, cwd=m.config.work_dir, stats=stats, rewrite_stdout_env=rewrite_env)
  File "C:\Miniconda\lib\site-packages\conda_build\utils.py", line 412, in check_call_env
    return _func_defaulting_env_to_os_environ('call', *popenargs, **kwargs)
  File "C:\Miniconda\lib\site-packages\conda_build\utils.py", line 392, in _func_defaulting_env_to_os_environ
    raise subprocess.CalledProcessError(proc.returncode, _args)
subprocess.CalledProcessError: Command '['cmd.exe', '/d', '/c', 'conda_build.bat']' returned non-zero exit status 2.
##[error]Cmd.exe exited with code '1'.
Finishing: Build recipe
```

**Comment by peterjc** [comment]
@conda-forge/help-c-cpp Ready for review?

If the Windows problem is obvious, I'd be happy to have that working from the start.

**Comment by peterjc** [comment]
Note to self, instructions on https://opensource.com/article/17/1/jove-lightweight-alternative-vim for setting the non standard delete/backspace key usage:

```
# DEL key
bind-to-key delete-next-character ^[[3~
# Backspace key
bind-to-key delete-previous-character ^?
```

Might be worth doing this in the package's configuration rather than ``~/.joverc``?

File: recipes/jove/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    - ncurses  # [not win]
+
+test:
+  commands:
+    - which jove
+
+about:
+  home: https://github.com/jonmacs/jove/
+  license: LicenseRef-JOVE
+  license_file: README
```
Comment (saraedum) [inline recipes/jove/meta.yaml:33]: Maybe you could extract the license into a separate file in the recipe and use that?

File: recipes/jove/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+  host:
+    - ncurses  # [not win]
+
+test:
+  commands:
+    - which jove
```
Comment (saraedum) [inline recipes/jove/meta.yaml:28]: Is there something like `jove --version` or `jove --help` that we could use to actually test that the binary has no missing dependencies somehow?

**Review by saraedum — CHANGES_REQUESTED** [review]

**Comment by saraedum** [comment]
> If the Windows problem is obvious, I'd be happy to have that working from the start.

It's not obvious to me at least. I would not worry about it I think. Having Linux and macOS is a good start.

File: recipes/jove/meta.yaml (hunk 2 above)
Comment (peterjc) [reply recipes/jove/meta.yaml:28]: Not that I could discover, sadly no. And once started, it seems to need user interaction to quit.

File: recipes/jove/meta.yaml (hunk 1 above)
Comment (peterjc) [reply recipes/jove/meta.yaml:33]: You think the text from the start of the README as a file in the recipe repository would suffice?

```
##########################################################################
# This program is Copyright (C) 1986-2002 by Jonathan Payne.  JOVE is    #
# provided by Jonathan and Jovehacks without charge and without          #
# warranty.  You may copy, modify, and/or distribute JOVE, provided that #
# this notice is included in all the source files and documentation.     #
##########################################################################
```

(Perhaps without the ``#`` signs)

**Comment by peterjc** [comment]
Thank you!

P.S. To anyone curious and not familiar with conda-forge, this is now here:

https://github.com/conda-forge/jove-feedstock

**Pull request merged**